src/OrchardCore.Modules/OrchardCore.Placements/Views/Admin/Edit.cshtml

```diff
@@ -0,0 +1,76 @@
+@model EditShapePlacementViewModel
+@{
+    var returnUrl = ViewData["returnUrl"]?.ToString();
+}
+
+<style asp-name="codemirror"></style>
+<script asp-name="codemirror" depends-on="admin" at="Foot"></script>
+<script asp-name="codemirror-addon-display-autorefresh" at="Foot"></script>
+<script asp-name="codemirror-addon-edit-closetag" at="Foot"></script>
+<script asp-name="codemirror-addon-selection-active-line" at="Foot"></script>
+<script asp-name="codemirror-mode-css" at="Foot"></script>
+<script asp-name="codemirror-mode-htmlmixed" at="Foot"></script>
+<script asp-name="codemirror-mode-javascript" at="Foot"></script>
+<script asp-name="codemirror-mode-xml" at="Foot"></script>
+
+<h1>@RenderTitleSegments(T["Edit Placement"])</h1>
+
+<form asp-action="Edit" asp-route-returnUrl="@ViewData["returnUrl"]" method="post">
+    @Html.HiddenFor(m => m.Creating)
+    @if (!Model.Creating)
+    {
+        @Html.HiddenFor(m => m.ShapeType)
+    }
+
+    <div class="form-group" asp-validation-class-for="ShapeType">
+        <label asp-for="ShapeType">@T["Shape type"]</label>
+        <input asp-for="ShapeType" asp-is-disabled="!Model.Creating" class="form-control" autofocus />
+        <span asp-validation-for="ShapeType"></span>
+        <span class="hint">@T["The type of the shape."]</span>
+    </div>
+
+    <div class="form-group" asp-validation-class-for="Nodes">
+        <label asp-for="Nodes">@T["Placement rules"]</label>
+        <textarea asp-for="Nodes" rows="10" class="form-control"></textarea>
+    </div>
+
+    <div class="form-group">
+        <div class="btn-group">
+            <button type="submit" name="submit" class="btn btn-primary">@T["Save"]</button>
+            <button type="button" class="btn btn-primary dropdown-toggle dropdown-toggle-split" data-reference="parent" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                <span class="sr-only">@T["Toggle Dropdown"]</span>
+            </button>
+            <div class="dropdown-menu">
+                <button class="dropdown-item" name="submit" type="submit" value="SaveAndContinue">@T["and continue"]</button>
+            </div>
+        </div>
+
+        @if (!Model.Creating)
+        {
+            <a asp-route-action="Delete" asp-route-area="OrchardCore.Placements" asp-route-shapeType="@Model.ShapeType" class="btn btn-danger" role="button" itemprop="UnsafeUrl RemoveUrl">@T["Delete"]</a>
+        }
+
+        @if (Url.IsLocalUrl(returnUrl))
+        {
+            <a class="btn btn-secondary" href="@returnUrl">@T["Cancel"]</a>
+        }
+    </div>
+</form>
+
+<script at="Foot">
+    $(function () {
+        var optionsTextArea = document.getElementById('@Html.IdFor(x => x.Nodes)');
+        // When part rendered by a flow part only the elements scripts are rendered, so the html element will not exist.
```

Comment:
This comment not relevant here (and trying not to put inline javascript in // comments so it doesn't get rendered in the browser - using `@* *@`) - but here you can just delete the comment